cmd/check.go

```diff
@@ -174,7 +174,7 @@ func runCheckCmd(cmd *cobra.Command, args []string) {
 		err = displayControlResults(ctx, executionTree)
 		utils.FailOnError(err)
 
-		if len(exportFormats) > 0 {
+		if len(exportFormats) > 0 && !utils.IsContextCancelled(ctx) {
```

Comment:
can you move `!utils.IsContextCancelled(ctx)` into the start of `exportCheckResult`